### PR TITLE
configure_nvme

### PR DESCRIPTION
**Pull request by hussamnasir**
Fix configure_nvme routine to make it generic
mount_point is now defaulted to "" and later reassigned to /mnt/{linux device name} just in case the user is using multiple NVMEs and does not provide a mount_point.

### CODE REVIEW

**Review by sajith — APPROVED** [review]
Looks plausible to me, although I actually have not tested this. :-)

I've left some comments inline. Would you like to address them and mark them as resolved, @hussamnasir?

File: fabrictestbed_extensions/fablib/component.py [hunk 1]
```diff
@@ -493,51 +493,74 @@ def get_type(self) -> str:
...
         Note this works but may be reorganized.
 
-        :param mount_point: The mount point in the filesystem. Default = /mnt/nvme_mount
+        :param mount_point: The mount point in the filesystem. Default = "" later reassigned to /mnt/{linux device name}
         :type mount_point: String
         """
         output = []
         try:
-            output.append(self.node.execute("sudo fdisk -l /dev/nvme*"), quiet=True)
+            device_pci_id = self.get_pci_addr()[0]
```
[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py:507]: although the docs say that get_pci_addr() return was a string, i was getting a list with one element which was why i used [0]

[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py:507]: i will double check this but i know why i may have been seeing an array while i was testing.

[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py:507]: I just double checked this and i get an array of length 1 as a return value.
image

File: fabrictestbed_extensions/fablib/component.py [hunk 2]
```diff
@@ -493,51 +493,74 @@ def get_type(self) -> str:
...
+            stdout, stderr = self.node.execute(
+                'basename `sudo ls -l /sys/block/nvme*|grep "'
+                + str(device_pci_id)
+                + "\"|awk '{print $9}'`",
+                quiet=True,
+            )
```
[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py]: double bracing is used such cases. https://realpython.com/python-f-strings/#braces. I have this fixed in the latest git commit to this branch

File: fabrictestbed_extensions/fablib/component.py [hunk 3]
```diff
@@ -493,51 +493,74 @@ def get_type(self) -> str:
...
+            if stderr != "":
+                output.append(
+                    "Cannot find NVME device name for PCI ID : " + device_pci_id
```
[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py]: I have this fixed in the latest git commit to this branch

File: fabrictestbed_extensions/fablib/component.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py:507]: I am not this is something that can be fixed in fablib as fablib just passes on the value obtained from the fimModel.

File: fabrictestbed_extensions/fablib/component.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sajith) [reply fabrictestbed_extensions/fablib/component.py]: Cool, thanks! I did not know that.

File: fabrictestbed_extensions/fablib/component.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sajith) [reply fabrictestbed_extensions/fablib/component.py:507]: True, and it really is FIM that I'm not truly familiar with. But FABlib could be accurate in docstrings and type signature though.  I will follow up separately.

Thank you for looking into this!

File: fabrictestbed_extensions/fablib/component.py [hunk 4]
```diff
@@ -493,51 +493,69 @@ def get_type(self) -> str:
...
+                    f"{mount_point}"
+                )
             )
-            output.append(self.node.execute(f"df -h {mount_point}"), quiet=True)
+            output.append(self.node.execute(f"df -h {mount_point}"))
         except Exception as e:
             print(f"config_nvme Fail: {self.get_name()}")
             # traceback.print_exc()
-            raise Exception(str(output))
+            raise Exception(str(output) + str(e))
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/component.py:558]: This error message also probably is a candidate for f-string-ifying, if you want, but I guess it is not a big deal.

**Comment by sajith** [comment]
@hussamnasir The changes look good to me.  The holdup is that I would like @paul-ruth to weigh in before merging this PR. :-)

File: fabrictestbed_extensions/fablib/component.py (hunk 4 above)
Comment (hussamnasir) [reply fabrictestbed_extensions/fablib/component.py:558]: Thats an error on my part. It should not be there. I will fix it tomorrow

**Review by sajith — APPROVED** [review]

**Comment by sajith** [comment]
@hussamnasir Merging, after seeking Paul's blessings, and updating the changelog entry. Sorry this took a while!

**Pull request merged**